src/iuse.cpp

```diff
@@ -8302,7 +8302,7 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
     }
     
     const inventory &crafting_inv = p->crafting_inventory();
-    if( !crafting_inv.has_amount( "water", 40 ) && !crafting_inv.has_amount( "water_clean", 40 ) ) {
+    if( !crafting_inv.amount_of( "water", 40 ) && !crafting_inv.amount_of( "water_clean", 40 ) ) {
```

Comment:
`amount_of` returns the available amount (a number, not a boolean) and its second parameter is not a number but a boolean. `!amount_of( "water", 40 )` is interpreted as `amount_of( "water", true ) != 0`, which only checks for at least one unit water.